Transactions - new Auditer Messages behaviour

jirmauritz opened the pull request:
- PerunConnection created to catch important events such as setSavepoint, releaseSavepoint, rolback
- Auditer Messages are stored in list of lists where inner lists represent one nested transaction
 - when a transaction is successful, all the Auditer Messages from last list are copied to the previous list
 - when a transaction is not successful, the last list is deleted

zlamalp commented:
test this please.

zlamalp commented:
Hi, tests are not passing since they use springTransactionManager instead of perunTransactionManager bean. I'm not sure if we will ditch the first one (without auditer) or not, since log() method is used in a code a lot and requires auditer when transaction starts not after (when is active). Switching bean definition in abstract classes of tests like this: ```@TransactionConfiguration(defaultRollback = true, transactionManager = "perunTransactionManager")
``` should be enough. Or you can somehow fix the log() method and recognize that auditer messages resource is not initialized (null).

jirmauritz commented:
Hi, I fixed the log method. I had this fix already on the testing branch, I don't know why I didn't merge it with this branch. Closing pull request and opening new, fixed, one.
